src/utils/DarCollectionUtils.js

```diff
@@ -167,36 +168,53 @@ export const extractDacDataAccessVotesFromBucket = (bucket, user, adminPage) =>
   )(votes);
 
   if(!adminPage) {
-    memberVotesArrays = filter(
-      (memberVotes) => includes(
-        user.dacUserId,
-        map(memberVote => memberVote.dacUserId)(memberVotes)
-      )
-    )(memberVotesArrays);
+    memberVotesArrays = filterVoteArraysForUsersDac(memberVotesArrays, user);
   }
-  memberVotesArrays = flatMap((memberVotes) => memberVotes)(memberVotesArrays);
-  return memberVotesArrays;
+  return flatten(memberVotesArrays);
 };
 
 //Gets rp votes from this bucket by members of this user's DAC
-//Note that filtering by DAC does not occur for users viewing throgh admin review page
+//Note that filtering by DAC does not occur for users viewing through admin review page
 export const extractDacRPVotesFromBucket = (bucket, user, adminPage) => {
   const votes = !isNil(bucket) ? bucket.votes : [];
-  let rpVotes = flow(
+  let rpVoteArrays = flow(
     map(voteData => voteData.rp),
     filter((rpData) => !isEmpty(rpData)),
     map(filteredData => filteredData.memberVotes)
   )(votes);
 
   if(!adminPage) {
-    rpVotes = filter(
-      memberVotes => includes(
-        user.dacUserId,
-        map(memberVote => memberVote.dacUserId)(memberVotes)
-      )
-    )(rpVotes);
+    rpVoteArrays = filterVoteArraysForUsersDac(rpVoteArrays, user);
   }
-  return flatMap(memberVotes => memberVotes)(rpVotes);
+  return flatten(rpVoteArrays);
+};
+
+
+//Gets final votes from this bucket by members of this user's DAC
+//Note that filtering by DAC does not occur for users viewing through admin review page
+export const extractDacFinalVotesFromBucket = (bucket, user, adminPage) => {
+  const { votes = {}, isRP } = bucket;
+  const targetAttr = isRP ? 'rp' : 'dataAccess';
+  let finalVoteArrays = map((voteObj) =>
+    !isEmpty(voteObj) ? voteObj[targetAttr].finalVotes : []
+  )(votes);
+
+  if(!adminPage) {
+    finalVoteArrays = filterVoteArraysForUsersDac(finalVoteArrays, user);
+  }
+  return flatten(finalVoteArrays);
+};
+
+// Applies filter to arrays of votes grouped by election and
+// only keeps arrays where at least one vote has the dacUserId of the provided user
+const filterVoteArraysForUsersDac = (voteArrays = [], user) => {
+  const userIdsOfVotes = (votes) => {
+    return map(vote => vote.dacUserId)(votes);
+  };
+
+  return filter(
+    voteArray => includes(user.dacUserId, userIdsOfVotes(voteArray))
+  )(voteArrays);
```

Comment:
I'm not clear on this method. The intention, I think, should be to filter on the user's DAC, not the user. The logic, as I understand it, should look something like this:

1. Find the user's DAC id(s)
2. Filter the votes for users who have the same DAC id(s)

Reply:
> With the current approach, since each array of vote objects has the same electionId for all votes, we see if any votes in the array have the dacUserId of the current user. If so, we conclude that the vote array is for an election for the dac of the current user and keep the votes in the array.

I'm comfortable with your current approach. I was incorrect to think the filtering would filter out votes from other users within the scope of the current election.